src/components/Courses/CourseForm.tsx

```diff
@@ -5,36 +5,139 @@ import {
     VStack,
     Flex,
     Checkbox,
+    Box,
+    Spacer,
 } from "@chakra-ui/react";
 import { Field, Form, Formik } from "formik";
 
 import NumInput from "@components/NumInput";
 import { SemesterBadges } from "@components/SemesterBadges";
+import Table from "@components/Table";
+import { useEffect, useMemo, useState } from "react";
 
 const CourseForm = (props) => {
-    const { handleSubmit, data, disabled, formId } = props;
+    const { handleSubmit, formId, data, disabled } = props;
+    const [numSections, setNumSections] = useState({
+        fall: data?.fall_sections?.length || 1,
+        spring: data?.spring_sections?.length || 1,
+        summer: data?.summer_sections?.length || 1,
+    });
+    const [offerings, setOfferings] = useState({
+        fall: data?.fall_sections.length > 0 || false,
```

Comment:
0 is a falsy value in js so this could be converted to:
`fall: !!data?.fall_sections?.length`